### PR TITLE
Added member status to the home page

### PR DESCRIPTION
**Pull request by Lukman-01**
## Description

 I added batch member status and if checked in status to the home page

MemberStatus

Your ENS/address: 0x186a761645f2A264ad0A655Fb632Ca99150803A9

### CODE REVIEW

**Comment by derrekcoleman** [comment]
Thank you for the PR, @Lukman-01!

First, let's make the PR description even nicer. I love your use of screenshot to show the implementation. Could you add a couple more screenshots for the other possible states? 
- Wallet not connected at all
- Wallet connected, not part of the Batch
- Wallet connected, part of the Batch, not checked in yet (you can mock this locally even though you already checked in in "production" on OP mainnet)
The issue you're solving is about providing a clear UX for the user based on these possible states, so that would be very useful information to help us understand how you're thinking about it.

Could you also update your PR description to link to issue #5 ? Not only is that useful context, it also makes it so that when this PR gets merged the matching issue is automatically closed as well.

**Comment by Lukman-01** [comment]
Thank you @derrekcoleman for the observation. Instead of doing screenshot for each implementation, I made a short clip demostrating each implementation. I hope it covers every implementation needed.

https://www.loom.com/share/4e7125956a8f481ebef43f1847e5d119?sid=466dfe05-23a5-4283-a8b8-e3498dad428c

kindly check the above link for the demostration.

**Comment by Lukman-01** [comment]
Good morning Sir @derrekcoleman, kindly review if there will be any changes I need to make.

File: packages/nextjs/app/page.tsx
Comment (phipsae) [inline packages/nextjs/app/page.tsx:1]: How about displaying just one status? If I'm not on the allowlist, is it necessary to inform me that I haven't checked in? 

Perhaps we could add a status indicating that the wallet isn’t connected, so users know they need to connect it to view their status, as @derrekcoleman suggested?

**Review by phipsae — CHANGES_REQUESTED** [review]
Thanks for the PR!!!

See below my comments.

**Comment by Lukman-01** [comment]
Good day @derrekcoleman, @phipsae, I have Implemented all cases required for the membership status. 





As for the last one(if a member but have not checkedIn), together with afformentioned ones above are demostrated in the clip below

https://www.loom.com/share/996b3e853d9b46e2ac2814569bdacc5c?sid=4dea6659-620b-482f-b18d-36a1a718852a

File: packages/nextjs/app/page.tsx [hunk 1]
```diff
@@ -1,74 +1,234 @@
...
+interface Props {
+  children: React.ReactNode;
+  className?: string;
+}
+
+interface StarknetReact {
+  connectedAddress: string;
+  isAllowListed: boolean;
+  isCheckedIn: boolean;
+  userContractAddress: string;
```
Comment (phipsae) [inline packages/nextjs/app/page.tsx:19]: Do we need userContractAddress or is connectedAddress enough?

File: packages/nextjs/app/page.tsx [hunk 2]
```diff
@@ -1,74 +1,234 @@
...
+    <Card>
+      <CardHeader className="pb-3">
+        <div className="flex items-center justify-between">
+          <CardTitle>Your Status</CardTitle>
+          <span className="rounded-full px-2 py-1 text-sm font-medium bg-green-100 text-green-800">Checked In</span>
+        </div>
+      </CardHeader>
+      <CardContent>
+        <p className="text-sm opacity-70">
+          Active with contract: {userContractAddress?.slice(0, 6)}...{userContractAddress?.slice(-4)}
```
Comment (phipsae) [inline packages/nextjs/app/page.tsx:127]: Here you might want to use the SE2 component `<Address>`

**Review by phipsae — CHANGES_REQUESTED** [review]
I like it, thanks for the implementation!

Just a couple of little comments to work on, shouldn't be too much. And then we merge.

**Comment by Lukman-01** [comment]
Thank you Sir @phipsae, corrections implemented.

File: packages/nextjs/app/page.tsx [hunk 3]
```diff
@@ -2,73 +2,222 @@
...
+      <CardHeader className="pb-3">
+        <div className="flex items-center justify-between">
+          <CardTitle>Your Status</CardTitle>
+          <span className="rounded-full px-2 py-1 text-sm font-medium bg-green-100 text-green-800">Checked In</span>
+        </div>
+      </CardHeader>
+      <CardContent>
+        <p className="text-sm opacity-70">
+          Active with wallet address: <Address address={connectedAddress} />
+        </p>
```
Comment (phipsae) [inline packages/nextjs/app/page.tsx:118]: There’s a hydration error occurring. Please make sure to always check your console for any potential errors. 
Screenshot 2024-11-03 at 10 41 45

**Review by phipsae — CHANGES_REQUESTED** [review]
There's just a small error that needs to be fixed before we can merge. 

Thanks!

**Comment by Lukman-01** [comment]
Thank you very much Sir, I will be more observant of the console now. The issue has been resolved.

resolve

**Comment by phipsae** [comment]
Perfect, thanks for your work! Let's merge.

**Pull request merged**